### PR TITLE
get rid of testdrive `publish` flag

### PR DESCRIPTION
**Pull request by brennanvincent**
it does not make sense to create Confluent Avro sources without using a schema registry.

Make it so we never try to publish schemas for the non-Confluent Avro case, and we always do for the Confluent case.

### CODE REVIEW

File: src/testdrive/src/action/kafka/ingest.rs [hunk 1]
```diff
@@ -45,6 +44,21 @@ pub struct IngestAction {
...
+impl IngestAction {
+    /// Whether the action causes a schema to be published
+    /// to CSR
+    pub fn publish(&self) -> bool {
+        match &self.format {
+            Format::Avro {
+                confluent_wire_format,
+                ..
+            } => *confluent_wire_format,
+            Format::Protobuf { .. } => false,
```
Comment (guswynn) [inline src/testdrive/src/action/kafka/ingest.rs:56]: proto can use csr too

**Review by guswynn — COMMENTED** [review]
looks good! Ill approve when its non-draft!

File: src/testdrive/src/action/kafka/ingest.rs (hunk 1 above)
Comment (brennanvincent) [reply src/testdrive/src/action/kafka/ingest.rs:56]: We never publish Proto schemas to the CSR in testdrive AFAICT.

I will let someone who understands the semantics of Protobuf and how it interacts with the CSR try to figure out why that's true; this PR is scoped only to Avro.

File: src/compute/src/server.rs [hunk 2]
```diff
@@ -363,7 +364,7 @@ impl<'w, A: Allocate> Worker<'w, A> {
             // reasoning, as we cannot be certain that peeks still exist at any other worker.
 
             // Having reduced our installed command history retaining no peeks (above), we should be able
-            // to use track down installed dataflows we can use as surrogates for requested dataflows (which
+            // to track down installed dataflows we can use as surrogates for requested dataflows (which
```
Comment (benesch) [inline src/compute/src/server.rs:367]: This changes seem fine but unrelated?

**Review by benesch — APPROVED** [review]
Good stuff!

File: src/testdrive/src/action/kafka/ingest.rs (hunk 1 above)
Comment (benesch) [reply src/testdrive/src/action/kafka/ingest.rs:56]: Protobuf schemas are split across multiple files. There's no easy way to have the equivalent of the `schema` option that we have for Avro-formatted records.

**Pull request merged**